pandas/core/frame.py

```diff
@@ -5058,7 +5065,12 @@ def _sanitize_column(self, value) -> tuple[ArrayLike, BlockValuesRefs | None]:
 
         if is_list_like(value):
             com.require_length_match(value, self.index)
-        return sanitize_array(value, self.index, copy=True, allow_2d=True), None
+
+        if not pa_version_under10p1 and isinstance(value, pa.lib._PandasConvertible):
+            dtype = ArrowDtype(value.type)
+        else:
+            dtype = None
+        return sanitize_array(value, self.index, dtype, copy=True, allow_2d=True), None
```

Comment:
Maybe related to my comment at https://github.com/pandas-dev/pandas/issues/56994#issuecomment-2179195600, but should the above logic moved _into_ `sanitize_array`?

Reply:
Yes, I just moved the `dtype `check in `sanitize_array`.

By the way I have now the following error in the CI :
```
Starting mysql service container
  /usr/bin/docker pull mysql:8.0.33
  Error response from daemon: Head "https://registry-1.docker.io/v2/library/mysql/manifests/8.0.33": received unexpected HTTP status: 503 Service Unavailable
```

Would you have any idea ?

Thanks in advance !